extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -0,0 +1,312 @@
+     * Newly created buckets (insert operations) are not cached
+     * as queue populator instances read from different kafka
+     * partitions and so don't need the configs for all buckets
+     * @returns {undefined}
+     */
+    _setMetastoreChangeStream() {
+        /**
+         * To avoid processing irrelevant events
+         * we filter by the operation types and
+         * only project the fields needed
+         */
+        const changeStreamPipeline = [
+            {
+                $match: {
+                    $or: [
+                        { operationType: 'delete' },
+                        { operationType: 'replace' },
+                        { operationType: 'update' },
+                    ]
+                }
+            },
+            {
+                $project: {
+                    '_id': 1,
+                    'operationType': 1,
+                    'documentKey._id': 1,
+                    'fullDocument._id': 1,
+                    'fullDocument.value.notificationConfiguration': 1
+                },
+            },
+        ];
+        this._metastoreChangeStream = new ChangeStream({
+            logger: this._logger,
+            collection: this._metastore,
+            pipeline: changeStreamPipeline,
+            handler: this._handleChangeStreamChangeEvent.bind(this),
+            throwOnError: false,
+            useStartAfter: semver.gte(this._mongoVersion, '4.2.0'),
+        });
+        // start watching metastore
+        this._metastoreChangeStream.start();
+    }
+
+    /**
+     * Sets up the MongoConfigManager by
+     * connecting to mongo and initializing the
+     * change stream
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    setup(cb) {
+        this._setupMongoClient(err => {
+            if (err) {
+                this._logger.error('An error occured while setting up mongo client', {
+                    method: 'MongoConfigManager.setup',
+                });
+                return cb(err);
+            }
+            try {
+                this._setMetastoreChangeStream();
+            } catch (error) {
+                this._logger.error('An error occured while establishing the change stream', {
+                    method: 'MongoConfigManager._setMetastoreChangeStream',
+                });
+                return cb(error);
+            }
+            return cb();
+        });
+    }
+
+    /**
+     * Get bucket notification configuration
+     *
+     * @param {String} bucket - bucket
+     * @return {Object|undefined} - configuration if available or undefined
+     */
+    async getConfig(bucket) {
+        const startTime = Date.now();
+        // return cached config for bucket if it exists
+        const cachedConfig = this._cachedConfigs.get(bucket);
+        if (cachedConfig) {
+            const delay = (Date.now() - startTime) / 1000;
+            onConfigManagerConfigGet(true, delay);
+            return {
```

Comment:
The old config manager used to store the config in zookeeper this way. To avoid having to migrate the old data stored in zookeeper or having to deal with the old vs new format i decided to keep the old one.